### ISSUE TITLE
Check for existing tmux session not working

### ISSUE THREAD

**Issue by rmonico** [danielshahaf/taskswamp#4]
Symptom: the program fail with error if there's no tmux server running. A workaround I using is before run the program do `tmux new-session -s any_session_name` (and kill the session after run the program). The problem happens because the check for existing task session checks just for output, not for non-zero return code (in this case the return code is 1, but I don't know if check this value is a good idea: maybe this return code be used for other kinds of errors.... I think would be better use has-session instead list-sessions, it's made for this situation and is in man page that it will return 1 in case the given session name doesn't exist (i.e. it should not be used for other errors).

**Comment by danielshahaf** [comment]
I agree with your analysis, but I can't reproduce the original problem. Could you try the following patch?

``` diff
commit 070000215056c78fdcb1e71a031063e22c51a059
Author: Daniel Shahaf <d.s@daniel.shahaf.name>
Date:   Wed Jun 29 07:38:25 2016 +0000

    bugfix: Check tmux session existence correctly when no tmux server is present.

    Fixes #4.

diff --git a/taskswamp b/taskswamp
index 54e1334..7ecd646 100755
--- a/taskswamp
+++ b/taskswamp
@@ -15,6 +15,8 @@ del THISDIR
 PROGNAME = os.path.basename(sys.argv[0])
 TASK = 'task' # path to the task binary
 TMUX = 'tmux' # path to the tmux binary
+EXIT_SUCCESS = 0
+DEVNULL = subprocess.DEVNULL if sys.version_info > (3, 3) else open(os.devnull, 'w+')

 @functools.lru_cache(maxsize=1)
 def taskdir():
@@ -40,7 +42,7 @@ def execve_to_system(args):

 def tmux_session_exists(sessionname):
     "Return TRUE if a tmux session SESSIONNAME is running."
-    return sessionname in set(
-        subprocess.check_output([TMUX, 'list-sessions', '-F', '#S']).decode().splitlines()
+    return (
+        EXIT_SUCCESS == subprocess.call([TMUX, 'has-session', '-t', sessionname], stderr=DEVNULL)
     )

```

> The problem happens because the check for existing task session checks just for output, not for non-zero return code

It does does check the exit code: it calls a function that returns if the exit code is 0 and raises otherwise.

https://docs.python.org/3/library/subprocess.html#subprocess.check_output

> [tmux has-session] will return 1 in case the given session name doesn't exist (i.e. it should not be used for other errors).

It returns 1 both for "can't connect to server" and for "server is running but has no session by this name".  I suppose we can still use it, though, can't we?

**Comment by danielshahaf** [comment]
Easier: `DEVNULL = getattr(subprocess, 'DEVNULL', open(…))`.
